pkgs/development/compilers/idris2/default.nix

```diff
@@ -0,0 +1,45 @@
+{ stdenv, fetchFromGitHub, makeWrapper
+, clang, chez
+}:
+
+# Uses scheme to bootstrap the build of idris2
+stdenv.mkDerivation {
+  name = "idris2";
+  version = "0.2.0-840e020";
+
+  src = fetchFromGitHub {
+    owner = "idris-lang";
+    repo = "Idris2";
+    rev = "840e020d8ccc332135e86f855ad78053ca15d603";
+    sha256 = "1l6pdjiglwd13pf56xwzbjzyyxgz48ypfggjgsgqk2w57rmbfy90";
+  };
+
+  strictDeps = true;
+  nativeBuildInputs = [ makeWrapper clang chez ];
+  buildInputs = [ chez ];
+
+  prePatch = ''
+    patchShebangs --build tests
+
+    # Do not run tests as part of the build process
+    substituteInPlace bootstrap.sh --replace "make test" "# make test"
```

Comment:
I added this because the build script of Idris2 does an automatic test. As far as I can tell, it's not common to run tests after builds, so I deactivated them.

I havn't found a good way to reproduce the test command as a separate phase, since it [depends on a few paths](https://github.com/idris-lang/Idris2/blob/e7d27bc46a8f4c12094732f38e566375812f18a6/bootstrap.sh#L49) that seem only to be available in the script.

I wanted to keep this derivation as close to the intended build process, since I suspect it's going to change, and I don't want to spend a lot of time rewriting this derivation every-time it happens.

Any suggestions welcome.

Reply:
I think I'd personally prefer running the tests as part of the build phase to not running them at all. Maybe upstream would be amenable to making `bootstrap.sh` take a phase to run (e.g. `stage1`, `stage2`, `test`) so that this could be split out into `checkPhase` without duplicating the script's logic?

Reply:
I added an upstream feature request to separate building from testing: https://github.com/idris-lang/Idris2/issues/125